lib/validate/net.go

```diff
@@ -22,6 +22,25 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// NetworkOverlap verifies that ipAddr is not in the range of the subnetCIDR
+func NetworkOverlap(ipAddr string, subnetCIDR string, errMsg string) error {
```

Comment:
Nit: Could you shorter form here since all arguments are of the same type:

```go
func NetworkOverlap(ipAddr, subnetCIDR, errMsg string) error {
```